homeassistant/components/publibike/sensor.py

```diff
@@ -0,0 +1,68 @@
+"""Support for PubliBike Public API for bike sharing in Switzerland."""
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PubliBikeDataUpdateCoordinator
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the PubliBike sensors."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([EBikeSensor(coordinator), BikeSensor(coordinator)])
+
+
+class PubliBikeSensor(CoordinatorEntity[PubliBikeDataUpdateCoordinator], SensorEntity):
+    """Representation of a PubliBike sensor."""
+
+    def __init__(self, coordinator):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.attributes = {}
```

Comment:
Attributes are not required.
To group all sesnor together we set a device info

```suggestion
        self._attr_device_info = DeviceInfo(
            name=coordinator.station.name,
            identifiers={(DOMAIN, coordinator.station.stationId)},
            entry_type=DeviceEntryType.SERVICE
        )
```